### ISSUE TITLE
extract local variable should define the variable using final/const not var

### ISSUE THREAD

**Issue by tjx666** [dart-lang/sdk#46536]
This tracker is for issues related to:

* Analyzer

**Describe the bug**

extract local variable should define the variable using final/const not var

![Jul-01-2021 15-53-52](https://user-images.githubusercontent.com/41773861/124087442-95e0ad80-da84-11eb-9308-e94c5a296f6f.gif)

**Expected behavior**

Normally, the express should be defined to final, because they are using only once.
Sometimes, like following example, should be defined to constant:

![image](https://user-images.githubusercontent.com/41773861/124088120-3931c280-da85-11eb-8c5d-91ffee39c689.png)

* Dart SDK Version (`dart --version`): 2.2.2
* Whether you are using Windows, MacOSX, or Linux (if applicable): MacOSX Big Sur 11.4

**Comment by srawlins** [comment]
Duplicate of https://github.com/dart-lang/sdk/issues/31701
